akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -253,9 +253,11 @@ object ByteString {
       else ByteString1(bytes, startIndex, Math.min(n, length))
 
     override def slice(from: Int, until: Int): ByteString = {
-      if (from <= 0 && until >= length) this // we can do < / > since we're Compact
+      if (from <= 0 && length <= until) this // we can do < / > since we're Compact
       else if (until <= from) ByteString1.empty
-      else ByteString1(bytes, startIndex + from, until - from)
+      else if (0 <= from && until <= bytes.length) ByteString1(bytes, startIndex + from, until - from)
+      else if (from < 0) ByteString1(bytes, startIndex, until)
+      else /*(until > bytes.length)*/ ByteString1(bytes, startIndex + from, bytes.length - from)
     }
```

Comment:
Thanks alot for your help and work! I've got two questions about your code!;

a) Do you assume `from < length` is satisfied when `ByteString1#slice(...)` is called? (I checked `if (length <= from) return empty` in `ByteString#slice(...)` on this [line](https://github.com/akka/akka/pull/21248/files#diff-c8291b63bef2cf0ccb0123d3e15e3833R166) to confirm the condition is met.) or we _CAN'T_ expect the condition?

b) In the third condition
```
else if (until <= length) ByteString1(bytes, startIndex + from, until - from)
```
Did you check `until <= length` using [length](https://github.com/monkey-mas/akka/blob/ea4ff374adf66a4ce51f6ee2b45313549f8f33c2/akka-actor/src/main/scala/akka/util/ByteString.scala#L206), seemingly the length of _sliced_ bytes? or do you mean `bytes.length`, the length of _original_ bytes here?

Based on your answers, I'll take a careful look again! :)